site/src/component/PPCOverlayTrigger/PPCOverlayTrigger.tsx

```diff
@@ -1,71 +1,98 @@
+  popoverContent,
+  children,
+  popupListener,
+  disabled = false,
+  anchor,
+  transform,
+}) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+  const open = Boolean(anchorEl);
 
-  const POPOVER_DELAY = 80;
+  const showPopover = (event: React.MouseEvent<HTMLElement>) => {
+    if (disabled) {
+      return;
+    }
 
-  const showPopover = () => {
-    setShowInfoPopover(true);
+    setAnchorEl(event.currentTarget);
     popupListener?.(true);
-    clearTimeout(showPopoverTimeout);
-    setShowPopoverTimeout(0);
   };
 
   const hidePopover = () => {
-    setShowInfoPopover(false);
+    setAnchorEl(null);
     popupListener?.(false);
-    clearTimeout(showPopoverTimeout);
-    setShowPopoverTimeout(0);
   };
 
-  const handleMouseMove = () => {
-    if (!showPopoverTimeout) return;
-    clearTimeout(showPopoverTimeout);
-    setShowPopoverTimeout(window.setTimeout(showPopover, POPOVER_DELAY));
-  };
+  const handleUnhover = (e: React.MouseEvent) => {
+    const relatedTarget = e.relatedTarget as Node | null;
+    const popoverContent = document.querySelector('.popover-paper');
 
-  const handleHover = () => {
-    if (document.querySelector('.course.sortable-fallback')) return;
-    if (disabled) return;
-    clearTimeout(showPopoverTimeout);
-    setShowPopoverTimeout(window.setTimeout(showPopover, POPOVER_DELAY));
-  };
-
-  const handleUnhover = (event: React.MouseEvent) => {
-    try {
-      const inTooltip = document.querySelector('.ppc-popover')?.contains(event?.relatedTarget as HTMLElement);
-      if (!inTooltip) hidePopover();
-    } catch {
+    if (!popoverContent || !relatedTarget || !popoverContent.contains(relatedTarget)) {
       hidePopover();
     }
   };
 
-  const popover = (
-    <Popover className="ppc-popover" id="ppc-popover" onMouseLeave={hidePopover}>
-      {popoverContent}
-    </Popover>
-  );
+  const clonedChild = React.cloneElement(children, {
+    onMouseEnter: (e: React.MouseEvent<HTMLElement>) => {
+      showPopover(e);
+      children.props.onMouseEnter?.(e);
+    },
+    onMouseLeave: (e: React.MouseEvent<HTMLElement>) => {
+      handleUnhover(e);
+      children.props.onMouseLeave?.(e);
+    },
+  });
 
   return (
-    <OverlayTrigger {...passedProps} show={showInfoPopover} overlay={popover}>
-      <div onMouseEnter={handleHover} onMouseLeave={handleUnhover} onMouseMove={handleMouseMove}>
-        {children}
-      </div>
-    </OverlayTrigger>
+    <div className="ppc-popover">
+      {clonedChild}
+      <Popover
+        open={open}
+        anchorEl={anchorEl}
+        onClose={hidePopover}
+        anchorOrigin={anchorMap[anchor]}
+        transformOrigin={transformMap[transform]}
+        slotProps={{
+          paper: {
+            className: 'popover-paper',
+            onMouseLeave: hidePopover,
+            sx: {
+              pointerEvents: 'auto',
+            },
+          },
+        }}
+        sx={{
+          pointerEvents: 'none',
+        }}
```

Comment:
Since these are all popovers that we can hover, I'd recommend something more like setting a className on `<Popover/>` itself, perhaps something like `.hoverable-popover`.

That way, you can also use _that_ as your more specific css selector for Paper (see previous comment) and not have to use `sx: { pointerEvents }`.

The change in the stylesheet would be something like:
```scss
.hoverable-popover {
  pointer-events: none;

  .MuiPaper-root {
    // current styles
    pointer-events: auto;
  }
}
```
then you could greatly simplify this:

```suggestion
        className="hoverable-popover"
        slotProps={{
          paper: { onMouseLeave: hidePopover },
        }}
```